### PR TITLE
[POAE7-2774] HashTable implementation of float and double

### PR DESCRIPTION
**Pull request by Deegue**
### What changes were proposed in this pull request?
1. Add float and double type support for `AggregationHashTable`.
2. Add related test cases.

### Why are the changes needed?
Implementation.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
UTs.

### CODE REVIEW

**Comment by jikunshang** [comment]
need to add some tests in CiderNewHashTableTest ?

File: cpp/src/cider/tests/CiderNewAggHashTableTest.cpp [hunk 1]
```diff
@@ -537,6 +540,260 @@ TEST_F(CiderNewAggHashTableTest, aggUInt64Test) {
...
+
+  float key1 = 1.1;
+  float key2 = 2.2;
+  float key3 = 1.1;
+
+  int32_t val1 = 10;
+  int32_t val2 = 20;
+  int32_t val3 = 30;
+
+  std::vector<int8_t> offset_vec{2, 8, 12, 16};
```
[the remark this answers is not in the document]
Comment (Deegue) [reply cpp/src/cider/tests/CiderNewAggHashTableTest.cpp:567]: Yes, `offset_vec[0]` is key offset and `offset_vec[1]` to `offset_vec[3]` are value offsets.
Added comments in the code.

File: cpp/src/cider/tests/CiderNewAggHashTableTest.cpp [hunk 2]
```diff
@@ -537,6 +540,260 @@ TEST_F(CiderNewAggHashTableTest, aggUInt64Test) {
...
+  // Row number   key    value
+  //     0         1.1       10
+  //     1         2,2       20
+  //     2         1.1       30
+
+  // key of HT: float
+  std::vector<SQLTypes> keys;
+  keys.push_back(SQLTypes::kFLOAT);
+
+  uint8_t key_len = 6;
```
[the remark this answers is not in the document]
Comment (Deegue) [reply cpp/src/cider/tests/CiderNewAggHashTableTest.cpp:565]: 1byte is_null(bool) + 1byte padding + 4bytes float
Added comments in the code.

File: cpp/src/cider/tests/CiderNewAggHashTableTest.cpp [hunk 3]
```diff
@@ -537,6 +540,260 @@ TEST_F(CiderNewAggHashTableTest, aggUInt64Test) {
...
+  float key2 = 2.2;
+  float key3 = 1.1;
+
+  int32_t val1 = 10;
+  int32_t val2 = 20;
+  int32_t val3 = 30;
+
+  std::vector<int8_t> offset_vec{2, 8, 12, 16};
+
+  // value of HT: SUM(int32)-int64 + COUNT(int32)-int32 + MIN(int32)-int32 +
```
[the remark this answers is not in the document]
Comment (Deegue) [reply cpp/src/cider/tests/CiderNewAggHashTableTest.cpp]: That's incorrect, revised.

**Comment by Deegue** [comment]
> need to add some tests in CiderNewHashTableTest ?

Added

**Review by jikunshang — APPROVED** [review]

**Pull request merged**